reposerver/cache/cache.go

```diff
@@ -189,10 +221,16 @@ func (c *Cache) SetManifests(revision string, appSrc *appv1.ApplicationSource, c
 		res.CacheEntryHash = hash
 	}
 
+	if log.IsLevelEnabled(log.DebugLevel) {
+		logEntryWithManifestCacheKeyFields(revision, appSrc, clusterInfo, namespace, trackingMethod, appLabelKey, appName, setReason).Debugf("setting manifests cache")
+	}
 	return c.cache.SetItem(manifestCacheKey(revision, appSrc, namespace, trackingMethod, appLabelKey, appName, clusterInfo), res, c.repoCacheExpiration, res == nil)
 }
 
-func (c *Cache) DeleteManifests(revision string, appSrc *appv1.ApplicationSource, clusterInfo ClusterRuntimeInfo, namespace string, trackingMethod string, appLabelKey string, appName string) error {
+func (c *Cache) DeleteManifests(revision string, appSrc *appv1.ApplicationSource, clusterInfo ClusterRuntimeInfo, namespace, trackingMethod, appLabelKey, appName, deleteReason string) error {
```

Comment:
This function is already having lots of parameters. I don't think it is a good idea to introduce an additional one just to be used in the log. It is better to log with the specific reason before calling this function.